esrally/log.py

```diff
@@ -57,7 +57,11 @@ def install_default_log_config():
         source_path = io.normalize_path(os.path.join(os.path.dirname(__file__), "resources", "logging.json"))
         with open(log_config, "w", encoding="UTF-8") as target:
             with open(source_path, "r", encoding="UTF-8") as src:
-                contents = src.read().replace("${LOG_PATH}", paths.logs())
+                # Ensure we have a trailing path separator as after LOG_PATH there will only be the file name
+                log_path = os.path.join(paths.logs(), "")
+                # the logging path might contain backslashes that we need to escape
+                log_path = io.escape_path(log_path)
```

Comment:
I wonder if we can avoid some of the special handling in the future by relying on the methods in the [pathlib](https://docs.python.org/3/library/pathlib.html#module-pathlib) module.

Reply:
I agree that this is unfortunate but the problem is that this string ends up in JSON. On Windows, a path ends up as the JSON string `C:\some\path` and when this string is read, the `\s` and `\p` are interpreted as escape sequences. This situation is similar to regex literals but for these we can use raw strings (literal prefixed by `r`) to avoid that.